closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -303,13 +315,15 @@ void updateAAD(byte[] src, int offset, int len) {
      * @return the number of bytes placed into the <code>out</code> buffer
      */
     int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
-        checkDataLength(ibuffer_enc.size(), len);
-
-        if (len > 0) {
-            // store internally until encryptFinal is called because
-            // spec mentioned that only return recovered data after tag
-            // is successfully verified
-            ibuffer_enc.write(in, inOfs, len);
+        synchronized(this) {
+            checkDataLength(ibuffer_enc.size(), len);
```

Comment:
`ibuffer_enc` may be `null` here; create an object if necessary like on line 340.

Reply:
Fixed.